Documentation/devel/distribution.md

```diff
@@ -0,0 +1,147 @@
+
+**NOTE:** I'm not sure if distribution is the best name for this concept or it should be called with other names like "image reference". So I'd like to hear your thoughts.
```

Comment:
`Distribution Points` looks good. I was also thinking to just `Distribution URIs`.